### ISSUE TITLE
Context as a method parameter in GraphQLApi class + Java Bean Validation input

### ISSUE THREAD

**Issue by raoua-eng** [quarkusio/quarkus#24183]
### Describe the bug

I upgraded the quarkus version to **2.7.3.Final** and I'm using Jakarta validation API to validate GraphQL inputs.  For instance, in this example my GraphQL input deal with the _Size_, _Email_, _Past_ and _NotBlank_ annotations:
```java
import java.time.LocalDate;

import javax.json.bind.annotation.JsonbDateFormat;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Past;
import javax.validation.constraints.Size;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class CreatePersonUseCaseInput {
    @NotBlank(message="'${validatedValue}' cannot be null")
    private String name;
    @Past(message="Date must be in past. found '${validatedValue}'")
    @JsonbDateFormat(value = "dd/MM/yyyy")
    private LocalDate dateOfBirth;
    @Email(message="email should be valid")
    private String email;
    @Size(max = 5, message = "'${validatedValue}' must be at least {max} characters long. Length found : '${validatedValue.length()}'")
    private String civility;
 
}
``` 
When I run a mutation from the GraphQL playground an exception is thrown saying :
```
ERROR [io.sma.graphql] (vert.x-worker-thread-0) SRGQL012000: Data Fetching Error: java.lang.IndexOutOfBoundsException: index (1) must be less than size (1)
```
In my mutation method I'm using the io.smallrye.graphql.api.Context as a parameter to handle the graphql request and I see it's the source of the error because when I remove it everything goes okay.
```java
public Class GraphQLResource{
   @Inject
    ObjectMapper objectMapper;

    @Mutation
    @Description("create person")
    public String createPerson(Context context, @Valid @Name("person") CreatePersonUseCaseInput input) throws Exception {
        System.out.println("Mutation Context : "+ context);
        /*
         * do something
         * */
        CreatePersonUseCaseBO personBO=new CreatePersonUseCaseBO("Raoua",null, "test@test", "M");
        CreatePersonUseCaseBO objectUpdated = MutationRequestHandler.getObject(context, "person", objectMapper, personBO);
       System.out.println(objectUpdated.toString());
        return "successful creation";
    }
}
```
I see it is related to the **size** annotation in combination with the Context as a parameter in the method.

### Expected behavior

The Context as a parameter in the method works well with input validation

### Actual behavior
```
ERROR [io.sma.graphql] (vert.x-worker-thread-0) SRGQL012000: Data Fetching Error: java.lang.IndexOutOfBoundsException: index (1) must be less than size (1)
    at graphql.com.google.common.base.Preconditions.checkElementIndex(Preconditions.java:1345)
    at graphql.com.google.common.base.Preconditions.checkElementIndex(Preconditions.java:1327)
    at graphql.com.google.common.collect.SingletonImmutableList.get(SingletonImmutableList.java:43)
    at io.smallrye.graphql.cdi.validation.BeanValidationException$RequestNodeBuilder.requestedArgument(BeanValidationException.java:66)
    at io.smallrye.graphql.cdi.validation.BeanValidationException$RequestNodeBuilder.build(BeanValidationException.java:54)
    at io.smallrye.graphql.cdi.validation.BeanValidationException.lambda$appendDataFetcherResult$0(BeanValidationException.java:35)
    at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
    at java.base/java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1603)
    at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
    at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
    at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
    at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
    at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
    at io.smallrye.graphql.cdi.validation.BeanValidationException.appendDataFetcherResult(BeanValidationException.java:36)
    at io.smallrye.graphql.execution.datafetcher.AbstractDataFetcher.get(AbstractDataFetcher.java:56)
    at graphql.execution.instrumentation.dataloader.DataLoaderDispatcherInstrumentation.lambda$instrumentDataFetcher$0(DataLoaderDispatcherInstrumentation.java:87)
    at graphql.execution.ExecutionStrategy.fetchField(ExecutionStrategy.java:279)
    at graphql.execution.ExecutionStrategy.resolveFieldWithInfo(ExecutionStrategy.java:210)
    at graphql.execution.ExecutionStrategy.resolveField(ExecutionStrategy.java:182)
    at graphql.execution.AsyncSerialExecutionStrategy.lambda$execute$1(AsyncSerialExecutionStrategy.java:43)
    at graphql.execution.Async.eachSequentiallyImpl(Async.java:80)
    at graphql.execution.Async.eachSequentially(Async.java:69)
    at graphql.execution.AsyncSerialExecutionStrategy.execute(AsyncSerialExecutionStrategy.java:38)
    at graphql.execution.Execution.executeOperation(Execution.java:159)
    at graphql.execution.Execution.execute(Execution.java:105)
    at graphql.GraphQL.execute(GraphQL.java:613)
    at graphql.GraphQL.parseValidateAndExecute(GraphQL.java:538)
    at graphql.GraphQL.executeAsync(GraphQL.java:502)
    at graphql.GraphQL.execute(GraphQL.java:433)
    at io.smallrye.graphql.execution.ExecutionService.execute(ExecutionService.java:126)
    at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLExecutionHandler.doRequest(SmallRyeGraphQLExecutionHandler.java:299)
    at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLExecutionHandler.handlePost(SmallRyeGraphQLExecutionHandler.java:113)
    at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLExecutionHandler.doHandle(SmallRyeGraphQLExecutionHandler.java:73)
    at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLAbstractHandler.handleWithIdentity(SmallRyeGraphQLAbstractHandler.java:66)
    at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLAbstractHandler.handle(SmallRyeGraphQLAbstractHandler.java:48)
    at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLAbstractHandler.handle(SmallRyeGraphQLAbstractHandler.java:23)
    at io.vertx.ext.web.impl.BlockingHandlerDecorator.lambda$handle$0(BlockingHandlerDecorator.java:48)
    at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
    at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
    at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
    at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
    at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
    at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
    at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
    at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
    at java.base/java.lang.Thread.run(Thread.java:834)

```
### How to Reproduce?

As a reproducer, please see [https://github.com/raoua-eng/Quarkus](url).
To reproduce the issue we need the following graphql input:
```
mutation{
  createPerson(
    person:{
      name:"test"
      email: "test@test.com"
      civility:"testttt"
     dateOfBirth:"26/12/2000"
    }
  )
}
```

### Output of `java -version`

11.0.1

### Quarkus version or git rev

2.7.3.Final

**Comment by phillip-kruger** [comment]
@jmartisk - this is a weird one, Context can be injected in the class, but in the method gives this error, only when using BV.

**Comment by gsmet** [comment]
Not sure if it helps: the constraint will make the method intercepted by the HV interceptor. Not sure it changes something for you.

**Comment by phillip-kruger** [comment]
Not sure, I assume the interception happens when it's called? We call the method using reflection, and we see if the context needs to be added to the parameters, and if so, we add it.

**Comment by jmartisk** [comment]
The problem is that we incorrectly compute the index where we the argument should be stored in the respective `FieldDefinition` - the `FieldDefinition` (that describes the operation for schema purposes) does not contain arguments of type `Context`, so when there's a `Context` argument on an operation, then the set of fields in the `FieldDefinition` is not the same as the set of Java method arguments (but we basically assume that it IS the same).
https://github.com/smallrye/smallrye-graphql/pull/1314 should fix it

**Comment by jmartisk** [comment]
So, this will be resolved by upgrading to SmallRye GraphQL 1.4.4 or 1.5.0, not sure yet what version we will release

**Comment by gsmet** [comment]
@jmartisk in any case, I think it would be a good idea to have something back portable to 2.7.
